network/p2p/p2p_setup.go

```diff
@@ -115,6 +117,52 @@ func (n *p2pNetwork) IsBadPeer(logger *zap.Logger, peerID peer.ID) bool {
 	return n.idx.IsBad(logger, peerID)
 }
 
+func (n *p2pNetwork) canAcceptNewConnection(lg *zap.Logger, peerID peer.ID) bool {
+	if n.idx == nil {
+		return false
+	}
```

Comment:
Could maybe just use `n.IsBadPeer` (instead of this and `n.idx.IsBad(lg, peerID)` below)